app/classes/content_filter.rb

```diff
@@ -0,0 +1,98 @@
+# filters which are applied in Query
+module ContentFilter
+
+# To add a new Content Filter:
+# * Indicates unnecessary for new observation filter -- it's done automatically.
+#
+#   Add tests, e.g., to TestUserContentFilter
+#   Supplement fixtures as needed by added tests
+#   Add a filter definition below, and add that definition to #filters below
+#   Add a new checkbox_val method in User
+#   In /config/locales/en.txt define text to be displayed before and next to
+#     check boxes, using prefs_obs_filters_has_images: as a model.
+# * Supplement _prefs_filters.html.erb as needed
+# * Add filter to AccountController#prefs_types & #update_content_filter
+# * For Observation filter, supplement Query::Initializers::ObservationFilters
+# * Supplement ApplicationController#show_index_of_objects as needed
+#   To filter another object, create a new initializer and include in
+#     appropriate searches.
+#
+# To be able to override the new filter in Advanced Searches, at least:
+#   Add tests, e.g., to TestAdvancedSearchFilters
+#   Supplement fixtures as needed by added tests
+#   In /config/locales/en.txt define text to be displayed before and next to
+#     radio boxes, using advanced_search_filter_has_images: as a model.
+# * Supplement _advanced_search_filters.html.erb as needed
+# * Supplement ObservationController#advanced_search_form as needed.
+# * Supplement ApplicationController#show_index_of_objects as needed.
+# * Supplement Query::RssLogBase as needed.
+#
+# To also add it as a Pattern Search
+#   Supplement PatternSearchTest
+#   Supplement classes/pattern_search/observation.rb or -- if it's not an
+#     Observation search -- add a new pattern_search file and class.
+#
+# There are probably other steps/files I've forgotten. JDC 2016-09-15
```

Comment:
@mo-nathan -
Thanks so much for taking the time to look at this so thoroughly.
- I'll check the Rubocop offenses (I suspect that most are preexisting) and fix at least the easier ones. 
- I agree with that the process should be simplified, but not needed for this PR. (I couldn't figure out how to make it any simpler; I thought it was best to push it out now to minimize future conflicts and make the functionality available. It can be refactored in the future if more content filters are added, and the additional filters should help make for a better refactoring.)